### PR TITLE
Added I2C code.

### PR DESCRIPTION
**Pull request by surendranadkarni**

### CODE REVIEW

File: src/linux/plt_i2c.c [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+int adapter;
+pthread_mutex_t i2c_mutex;
+
+int i2c_init(int dev_id) 
+{
+    char adaptername[20];
+
+    pthread_mutex_lock(&i2c_mutex);
+    snprintf(adaptername, 19, "/dev/i2c-%d", dev_id);
```
Comment (surendranadkarni) [inline src/linux/plt_i2c.c]: add following in platform/pvt header
#define PLT_I2C_CFG_GET_DEVICE_FILE(dev_id) ((dev_id==0)?"/dev/i2c0":NULL)

File: src/linux/plt_i2c.c [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    return MAP_I2C_CONTEXT_TO_PLT_HND(adapter);
+}
+
+int i2c_term(void) {
+    close(adapter);
+    return 0;
+}
+
+int i2c_write(UCHAR dev, UCHAR reg, UCHAR* data, int len) {
+    int ret;
```
Comment (surendranadkarni) [inline src/linux/plt_i2c.c]: i2c apis shall use handle not fd directly
Kindly check SPI function ```plt_spi_transfer(plt_handle hnd, uint08 const *tx, uint08 const *rx, int len)```

Comment (surendranadkarni) [reply src/linux/plt_i2c.c]: your api should look like
```
plt_status i2c_write(plt_handle hnd, uint08 dev_addr, uint08 reg_addr, uint08 const *data, int len)
```

File: src/linux/plt_i2c_pvt.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+typedef unsigned long   uint32;
+typedef unsigned short  uint16;
+typedef unsigned char   uint08;
+typedef void*           plt_handle;
+typedef int             plt_status;
+typedef unsigned char   bool;
+
+
+#define false   (0)
```
Comment (surendranadkarni) [inline src/linux/plt_i2c_pvt.h]: include <stdbool.h>instead

File: src/linux/plt_i2c_pvt.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+#define true    (!false)
+#define plt_assert(condn) assert(condn)
```
Comment (surendranadkarni) [inline src/linux/plt_i2c_pvt.h]: plt_assert -> use create defines.h and add PLT_ASSERT

File: src/linux/plt_i2c_pvt.h [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+#include <linux/i2c-dev.h>
+#include <linux/i2c.h>
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <pthread.h>
+
+
+typedef unsigned long   uint32;
+typedef unsigned short  uint16;
```
Comment (surendranadkarni) [inline src/linux/plt_i2c_pvt.h]: why to create same typedefs at multiple location

File: inc/plt_i2c.h [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+#include <stdlib.h>
+#include <linux/i2c-dev.h>
+#include <linux/i2c.h>
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <pthread.h>
+
+
+typedef unsigned long   uint32;
```
Comment (surendranadkarni) [inline inc/plt_i2c.h]: why to create same typedefs at multiple location

**Review by saurabhnamjoshi — COMMENTED** [review]
I2C code modified, as per review done by you.
Please go through it & let me know.

File: inc/datatypes.h [hunk 7]
```diff
@@ -31,6 +31,7 @@ typedef unsigned long   uint32;
 typedef unsigned short  uint16;
 typedef unsigned char   uint08;
 
+
```
Comment (surendranadkarni) [inline inc/datatypes.h]: we shall not add extra spaces

File: src/linux/plt_i2c.c [hunk 8]
```diff
@@ -0,0 +1,176 @@
+
+#define MAP_I2C_CONTEXT_TO_PLT_HND(context) ((plt_handle) context)
+
+#define MAP_PLT_HND_TO_I2C_CONTEXT(hnd) ((i2c_context*) hnd)
+
+pthread_mutex_t i2c_mutex;
```
Comment (surendranadkarni) [inline src/linux/plt_i2c.c]: mutex shall be part of i2c context

File: src/linux/plt_i2c.c [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+    return ret;
+}
+
+plt_status i2c_read(plt_handle hnd, uint08 dev, uint08 reg, uint08* data, int len) {
+    plt_status ret;
+    struct i2c_rdwr_ioctl_data i2c_data;
+    struct i2c_msg msg[2];
+    char tmp[2];
+    i2c_context *context = MAP_PLT_HND_TO_i2c_CONTEXT(hnd);
+            plt_assert(context != NULL);
```
Comment (surendranadkarni) [inline src/linux/plt_i2c.c]: alignment

File: src/linux/plt_i2c.c [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+
+    i2c_data.msgs = msg;
+    i2c_data.nmsgs = 2;     // two i2c_msg
+
+    tmp[0] = reg;
+    i2c_data.msgs[0].addr = dev >> 1;
+    i2c_data.msgs[0].flags = 0;     // write
+    i2c_data.msgs[0].len = 1;       // only one byte
+    i2c_data.msgs[0].buf = (__u8 *)tmp;
```
Comment (surendranadkarni) [inline src/linux/plt_i2c.c]: plz use uint8. it makes it more readable

File: src/linux/plt_i2c.c (hunk 9 above)
Comment (surendranadkarni) [reply src/linux/plt_i2c.c]: MAP_PLT_HND_TO_i2c_CONTEXT
or
 MAP_PLT_HND_TO_I2C_CONTEXT

File: test_app/test_i2c.c [hunk 11]
```diff
@@ -0,0 +1,33 @@
+#include <stdio.h>
```
Comment (surendranadkarni) [inline test_app/test_i2c.c:1]: we might have a test app for some eeprom

File: src/linux/plt_i2c.c [hunk 12]
```diff
@@ -0,0 +1,186 @@
+#include <plt_i2c_pvt.h>
+
+#define MAP_I2C_CONTEXT_TO_PLT_HND(context) ((plt_handle) context)
+
+#define MAP_PLT_HND_TO_I2C_CONTEXT(hnd) ((i2c_context*) hnd)
+
+
+plt_handle i2c_init(int dev_id)
```
Comment (surendranadkarni) [inline src/linux/plt_i2c.c]: plz check spi cod ```plt_handle plt_spi_new_device(int dev_id)```

File: src/linux/plt_i2c.c [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+
+plt_handle i2c_init(int dev_id) 
+{
+    char *i2c_dev = PLT_SPI_CFG_GET_DEVICE_FILE(dev_id);
+    i2c_context *context = (i2c_context*) malloc(sizeof(i2c_context));
+    pthread_mutex_lock(&context->i2c_mutex);
+    context->fd = open(i2c_dev, O_RDWR);
+    if (adapter < 0) {
+        perror(i2c_dev);
+        exit(-1);
```
Comment (surendranadkarni) [inline src/linux/plt_i2c.c]: return NULL

File: src/linux/plt_i2c.c [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+
+    ret = 0;
+out:
+    pthread_mutex_unlock(&context->i2c_mutex);
+    free(buffer);
+
+    return ret;
+}
+
+plt_status i2c_read(plt_handle hnd, uint08 dev, uint08 reg, uint08* data, int len)
```
Comment (surendranadkarni) [inline src/linux/plt_i2c.c]: ``` i2c_read_byte and i2c_read_buffer```

File: src/linux/plt_i2c_pvt.h [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+#include <linux/i2c.h>
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <plt_i2c.h>
+typedef struct __tag_i2c_context
+{
+    int fd;
+    pthread_mutex_t i2c_mutex;
```
Comment (surendranadkarni) [inline src/linux/plt_i2c_pvt.h]: clock can be part of context

**Pull request merged**